### PR TITLE
federation service controller: fixing a bug so that existing services are created in newly registered clusters

### PR DESCRIPTION
**Pull request by mfanjie**
A defect on federation service controller.
Steps to recreate:
1. boot federation control plane
2. create a service and then register a new cluster

Root cause:
the right sequence should be  

```
servicesToUpdate = s.serviceCache.allServices()
```

then

```
s.updateAllServicesToCluster(servicesToUpdate, newCluster) 
```

then

```
servicesToUpdate = s.updateDNSRecords(servicesToUpdate, newClusters)
```

Now the first two lines' sequence is on the contrary, so when updateDNSRecords return no error, the service will be removed from  servicesToUpdate, and updateAllServicesToCluster get nothing to update.

This PR make the call sequence correct.

[Analytics]()

### CODE REVIEW

**Comment by mfanjie** [comment]
@quinton-hoole please review if this should be put to 1.3, thanks.

**Comment by mfanjie** [comment]
@nikhiljindal I know it is in critical time, but I think this PR is worthy and safe to merge. Without this fix, the existing services cannot be created in new registered clusters.

File: federation/pkg/federation-controller/service/servicecontroller.go [hunk 1]
```diff
@@ -638,7 +638,7 @@ func (s *ServiceController) updateAllServicesToCluster(services []*cachedService
 	cluster, ok := s.clusterCache.clientMap[clusterName]
 	if ok {
 		for _, cachedService := range services {
-			appliedState := cachedService.appliedState
+			appliedState := cachedService.lastState
```
Comment (nikhiljindal) [inline federation/pkg/federation-controller/service/servicecontroller.go]: same. can you explain why this change?
Will also be good to add a comment here in code explaining why we are using lastState and not appliedState.

File: federation/pkg/federation-controller/service/servicecontroller.go [hunk 2]
```diff
@@ -235,7 +235,7 @@ func (s *ServiceController) Run(workers int, stopCh <-chan struct{}) error {
 	}
 	go wait.Until(s.clusterEndpointWorker, time.Second, stopCh)
 	go wait.Until(s.clusterServiceWorker, time.Second, stopCh)
-	go wait.Until(s.clusterSyncLoop, clusterSyncPeriod, stopCh)
```
Comment (nikhiljindal) [inline federation/pkg/federation-controller/service/servicecontroller.go]: Why this change?

**Comment by nikhiljindal** [comment]
ok to test

**Comment by nikhiljindal** [comment]
cc @kubernetes/sig-cluster-federation 

Also cc @madhusudancs since he is writing services e2e. Will be good to include this case there.

**Comment by nikhiljindal** [comment]
Added to 1.3 milestone

**Comment by mfanjie** [comment]
The lastState is assigned when service controller catch the service event, the appliedState is assigned after service controller process the event. When a new cluster added, a service that just be created but not be processed by service controller should also be applied to that cluster. So lastState should be used.

clusterSyncPeriod to second before the cluster sync loop interval is 10 minutes which is too long for users to wait for all services to apply to this cluster it should be 1 second interval like other loops.

**Comment by nikhiljindal** [comment]
thanks for the explanation @mfanjie 
Updating the title and correcting the release note label and readding the LGTM tag

**Comment by piosz** [comment]
@nikhiljindal why this is P1?
